.github/workflows/build.yml

```diff
@@ -35,6 +35,50 @@ on:
         type: boolean
 
 jobs:
+  load-catalog:
+    runs-on: ubuntu-latest
+    name: Load Test Catalog
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v4
+        with:
+          ref: 'test-catalog'
+          persist-credentials: false
+          fetch-depth: 100  # Needs to be large enough to ensure we can fetch N days ago
+
+      - name: Checkout catalog at earlier date
+        run: |
+          SHA=$(git rev-list -1 --before 7.days.ago origin/test-catalog)
+          echo $SHA
+          git switch --detach $SHA
+
+      - name: Checkout trunk
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          path: trunk
+
+      - name: Setup Python
+        uses: ./trunk/.github/actions/setup-python
```

Comment:
why does it require "./trunk" ?

Reply:
It is because I have checked out the main repo into a "trunk" directory https://github.com/apache/kafka/pull/17654/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R59.

I was running into issues with two `actions/checkout@v4` steps using the same destination directory, so this was my solution.

Let me try inverting this and checking `test-catalog` into its own directory. That would let me simply the `setup-python` action

Reply:
Ok, looks like this worked. `test-catalog` is now checked out into its own directory and the main repo is checked out in the usual working directory.